include/fmt/chrono.h

```diff
@@ -405,13 +405,33 @@ inline size_t strftime(wchar_t* str, size_t count, const wchar_t* format,
 
 FMT_END_DETAIL_NAMESPACE
 
-template <typename Char, typename Duration>
-struct formatter<std::chrono::time_point<std::chrono::system_clock, Duration>,
+template <typename Char, typename Rep, typename Period>
+struct formatter<std::chrono::time_point<std::chrono::system_clock, std::chrono::duration<Rep, Period>>,
                  Char> : formatter<std::tm, Char> {
   template <typename FormatContext>
   auto format(std::chrono::time_point<std::chrono::system_clock> val,
               FormatContext& ctx) -> decltype(ctx.out()) {
-    std::tm time = localtime(val);
+    std::tm time;
+    auto epoch = val.time_since_epoch();
+    auto seconds = std::chrono::duration_cast<std::chrono::seconds>(epoch);
+    auto subseconds = std::chrono::duration_cast<std::chrono::duration<Rep, Period>>(epoch - seconds);
+
+    if (subseconds < subseconds.zero()) {
+      time = localtime(val - std::chrono::seconds{1});
+      subseconds = std::chrono::seconds{1} + subseconds;
```

Comment:
nit: `std::chrono::seconds{1}` -> `std::chrono::seconds(1)` in two places for consistency with the rest of the library.